bob/build.go

```diff
@@ -3,30 +3,67 @@ package bob
 import (
 	"context"
 	"errors"
-
+	"fmt"
+	"github.com/benchkram/bob/bob/bobfile"
 	"github.com/benchkram/bob/bob/playbook"
+	"github.com/benchkram/bob/pkg/nix"
+	"github.com/benchkram/bob/pkg/usererror"
 	"github.com/benchkram/errz"
+	"strings"
 )
 
 var (
 	ErrNoRebuildRequired = errors.New("no rebuild required")
 )
 
-// Build a task and it's dependecies.
-func (b *B) Build(ctx context.Context, taskname string) (err error) {
+// Build a task and it's dependencies.
+func (b *B) Build(ctx context.Context, taskName string) (err error) {
 	defer errz.Recover(&err)
 
-	aggregate, err := b.Aggregate()
+	ag, err := b.Aggregate()
+	errz.Fatal(err)
+
+	b.PrintVersionCompatibility(ag)
+
+	workingDirectoryBobFile, err := bobfile.BobfileRead(ag.Dir())
```

Comment:
`BobfileRead()`  is a expensice operation. Could be avoided here as it has already been read.